docs/static/persistent-queues.asciidoc

```diff
@@ -42,7 +42,7 @@ TIP: If you want to define values for a specific pipeline, use <<multiple-pipeli
 
 `queue.type`:: Specify `persisted` to enable persistent queues. By default, persistent queues are disabled (default: `queue.type: memory`).
 `path.queue`:: The directory path where the data files will be stored. By default, the files are stored in `path.data/queue`. 
-`queue.page_capacity`:: The maximum size of a queue page in bytes. The queue data consists of append-only files called "pages". The default size is 64mb. Changing this value is unlikely to have performance benefits.
+`queue.page_capacity`:: The maximum size of a queue page in bytes. The queue data consists of append-only files called "pages". The default size is 64mb. Changing this value is unlikely to have performance benefits. You can change the page capacity of existing queue. The new size will apply to the new page.
```

Comment:
This is confusing to me.  We say that "Changing this value is unlikely to have performance benefits," but then we give more info how to change it, so it seems like a mixed message. 

Are there _other_ benefits that we should call out?

Reply:
One reason (likely not the only one):

Because each pipeline needs to have 1-2 pages open at any given time (the head that it is writing to, and the tail that it is reading from, which are often the same as each other), and doing so entails loading the page's file as memory-mapped, a logstash with many PQ-enabled pipelines (read:hundreds) can end up using a LOT of off-HEAP memory (64MB * 100 => 6GB). Reducing the page size can be a way to limit just how much memory is active per pipeline, which in turn allows a greater number of pipelines.

Reply:
we believe the page size, 64mb, is a good fit for most of the users, but at the same time, we do not stop them from tunning it to fit their situation. Here, I want to give a note that this value is changeable with an existing queue. Ry's example goes very deep about how PQ works inside. Probably, `queue.page_capacity` session is not the right place to share these details.